com.civicrm.hrjobroles/tests/phpunit/CRM/Hrjobroles/Import/Parser/HrJobRolesTest.php

```diff
@@ -0,0 +1,125 @@
+    ];
+    $this->runImport($importParams, 'assert_fail');
+  }
+
+  function testImportWithValidOptionValues() {
+    // create contact
+    $contactParams = ['first_name'=>'walter', 'last_name'=>'white'];
+    $contactID = $this->individualCreate($contactParams);
+
+    // create contract
+    $contract = $this->createJobContract($contactID, date('Y-m-d', strtotime('-14 days')));
+
+    // run importer
+    $importParams = [
+      'job_contract_id' => $contract->id,
+      'title' => 'test import role',
+      'location' => 'amman',
+      'hrjc_region' => 'south amman',
+      'hrjc_role_department' => 'amman devs',
+      'hrjc_level_type' => 'guru'
+    ];
+    $this->runImport($importParams);
+
+    $roleEntity = $this->findRoleByProperty($importParams['title'], 'title');
+    $this->assertEquals($importParams['title'], $roleEntity->title);
+    $this->assertEquals($importParams['location'], $roleEntity->location);
+    $this->assertEquals($importParams['hrjc_region'], $roleEntity->region);
+    $this->assertEquals($importParams['hrjc_role_department'], $roleEntity->department);
+    $this->assertEquals($importParams['hrjc_level_type'], $roleEntity->level_type);
+  }
+
+  function testImportWithInvalidOptionValues() {
+    // create contact
+    $contactParams = ['first_name'=>'walter', 'last_name'=>'white'];
+    $contactID = $this->individualCreate($contactParams);
+
+    // create contract
+    $contract = $this->createJobContract($contactID, date('Y-m-d', strtotime('-14 days')));
+
+    // run importer
+    $importParams = [
+      'job_contract_id' => $contract->id,
+      'title' => 'test import role2',
+      'location' => 'amman',
+      'hrjc_region' => 'southhggh ammandshhghg',
+      'hrjc_role_department' => 'amman devs',
+      'hrjc_level_type' => 'guru'
+    ];
+    $this->runImport($importParams, 'assert_fail');
+  }
+
+  function testImportWithEmptyOptionValues() {
+    // create contact
+    $contactParams = ['first_name'=>'walter', 'last_name'=>'white'];
+    $contactID = $this->individualCreate($contactParams);
+
+    // create contract
+    $contract = $this->createJobContract($contactID, date('Y-m-d', strtotime('-14 days')));
+
+    // run importer
+    $importParams = [
+      'job_contract_id' => $contract->id,
+      'title' => 'test import role3',
+      'location' => '',
+      'hrjc_region' => '',
+      'hrjc_role_department' => '',
+      'hrjc_level_type' => ''
+    ];
+    $this->runImport($importParams);
+
+    $roleEntity = $this->findRoleByProperty($importParams['title'], 'title');
+    $this->assertEquals($importParams['title'], $roleEntity->title);
+  }
+
+  private function runImport($params, $code = 'assert_success')  {
+    $fields = array_keys($params);
+    $values = array_values($params);
+    $importObject = new CRM_Hrjobroles_Import_Parser_HrJobRoles($fields);
+    $importObject->init();
+    ($code == 'assert_success') ? $code = CRM_Import_Parser::VALID : $code = CRM_Import_Parser::ERROR;
```

Comment:
I think this $code parameter and how you're asserting things here looks weird. What if you return the value from $importObject->import and do the assertion in the test methods? This will clearly separate the responsibilities:
The test method will do the assertions and the runImport will only, as the name says, run the Import